socorro/external/legacy_es/crashstorage.py

```diff
@@ -0,0 +1,689 @@
+
+    :param dict src: the source document with "processed_crash" key
+    :param dict crash_document: the document to fill
+    :param list fields: the list of fields in super search fields
+    :param set all_keys: the list of valid keys
+
+    """
+    for field in fields.values():
+        # There are some fields that aren't indexable--skip those
+        if not is_indexable(field):
+            continue
+
+        src_key = get_source_key(field)
+        value = glom.glom(src, src_key, default=None)
+        if value is None:
+            continue
+
+        # Fix values so they index correctly
+        storage_type = field.get("type", field["storage_mapping"].get("type"))
+
+        if (
+            storage_type == "multi_field"
+            and glom.glom(field, "storage_mapping.fields.full.type", default="")
+            == "string"
+        ):
+            storage_type = "string"
+
+        if storage_type == "string":
+            analyzer = field.get("analyzer", field["storage_mapping"].get("analyzer"))
+            if analyzer == "keyword":
+                value = fix_keyword(value, max_size=MAX_KEYWORD_FIELD_VALUE_SIZE)
+            else:
+                value = fix_string(value, max_size=MAX_STRING_FIELD_VALUE_SIZE)
+
+        elif storage_type == "integer":
+            value = fix_integer(value)
+            if value is None:
+                continue
+
+        elif storage_type == "long":
+            value = fix_long(value)
+            if value is None:
+                continue
+
+        elif storage_type == "date":
+            value = fix_datetime(value)
+            if value is None:
+                continue
+
+        for dest_key in get_destination_keys(field):
+            if dest_key in all_keys:
+                glom.assign(crash_document, dest_key, value, missing=dict)
+
+
+class LegacyESCrashStorage(CrashStorageBase):
+    """Indexes documents based on the processed crash to Elasticsearch."""
+
+    # These regex will catch field names from Elasticsearch exceptions. They
+    # have been tested with Elasticsearch 1.4.
+    field_name_string_error_re = re.compile(r"field=\"([\w\-.]+)\"")
+    field_name_number_error_re = re.compile(r"\[failed to parse \[([\w\-.]+)]]")
+    field_name_unknown_property_error_re = field_name_number_error_re
+
+    def __init__(
+        self,
+        url="http://localhost:9200",
+        index="socorro%Y%W",
+        index_regex=r"^socorro[0-9]{6}$",
+        retention_policy=26,
+        doctype="crash_reports",
+        metrics_prefix="processor.es",
+        timeout=30,
+        shards_per_index=10,
+    ):
+        super().__init__()
+
+        self.client = self.build_client(url=url, timeout=timeout)
+
+        # Create a MetricsInterface that includes the base prefix plus the prefix passed
+        # into __init__
+        self.metrics = markus.get_metrics(
+            build_prefix(METRICS.prefix, metrics_prefix),
+            filters=list(METRICS.filters),
+        )
```

Comment:
yeah, we'll be testing `ELASTICSEARCH_MODE=PREFER_NEW`, but right now those tests are expected to fail so I didn't add them to CI

Reply:
`metrics_prefix` here doesn't use the key from `settings.CRASH_DESTINATION` it uses the default value for the class, which is `"processor.es"` for both `ESCrashStorage` and `LegacyESCrashStorage`, and now that I've set up testing for PREFER_NEW it passes as-is.

Reply:
I double-checked this because it used to be composed so I was puzzled.

`metrics_prefix` has a default `processor.es` value in the classes. However, it's picking up the value set in options in `socorro/mozilla_settings.py` for both ES crash storage classes. If we need to change it in the future, we should change it in `socorro/mozilla_settings.py`.

This is probably fine. I'm wondering whether we should add an additional tag for which ES version it is. Do we want to be able to distinguish between metrics from one ES class from the other?